crates/agglayer-aggregator-notifier/src/sp1/mod.rs

```diff
@@ -48,21 +50,33 @@ where
 
         async move {
             spawn_blocking(move || {
-                prover.prove(
-                    &proving_key,
-                    stdin,
-                    Default::default(),
-                    Default::default(),
-                    Default::default(),
-                )
+                prover
+                    .prove(
+                        &proving_key,
+                        stdin,
+                        Default::default(),
+                        Default::default(),
+                        Default::default(),
+                    )
+                    .map_err(|error| {
+                        let error_description = error.to_string();
+                        if let Ok(error) = error.downcast::<ProofError>() {
```

Comment:
Maybe the `AggregatorProver` should specify the error state as an associated type or something like that to avoid `downcast` here. Seems we are trying to recover some type information at runtime even though it could be propagated at compilation time.

Reply:
The downcast is due to the fact that the `SP1Prover` returns an `anyhow::Error` which can be a `ProofError` from the `pessimistic-proof` program. We have no choice than `downcasting` the error to try getting more context.
I will put a TODO when we have time to check for a better solution.